samcli/commands/check/resources/graph.py

```diff
@@ -0,0 +1,50 @@
+"""
+Class for graph. All data is stored in the graph directly, or within nodes that are stored in the graph
+"""
+from typing import List, Optional, Union
+
+from samcli.commands.check.resources.lambda_function_pricing import LambdaFunctionPricing
+from samcli.commands.check.resources.lambda_function import LambdaFunction
+from samcli.commands.check.resources.warning import CheckWarning
+
+
+class CheckGraph:
+    """
+    A graph object stores four main item types.
+    - entry_points are all possible entry points to the application based on the template.
+    - resources_to_analyze are all resources that need to be directly checked for bottle
+      neck issues (just lambda functions for now).
+    - green/yellow/red/red_burst_warnings are the four different warning types that
+      the user can be presented
+    - lambda_function_pricing_info will contain the user entered data for the lambda
+      pricing questions
+    """
+
+    entry_points: List[Union[LambdaFunction]]
+    resources_to_analyze: List[Union[LambdaFunction]]
```

Comment:
AFAIR, `Union` is used if you have 2 different types that might be assigned to the variable. If so why we have a `Union[LambdaFunction]` which suggest to have single type (`LambdaFunction`)?

Reply:
Once more resources are added to the project, `Union` will be needed there. I just preemptively put it there, since it will be needed very soon.

Reply:
Will this grow to a long list of unions once we add more resources? 
Can there be a parent class or something which collectively represents these resources? not only for type checking, but also for exposing a common interface.

Reply:
Yes. `TemplateResource` is the parent class to `LambdaFunction` and all future resources. I will use that for all future type hints involving many resources

Reply:
Ok. I'll leave it for now.